packages/solidstart/src/vite/types.ts

```diff
@@ -0,0 +1,94 @@
+export type SourceMapsOptions = {
+  /**
+   * If this flag is `true`, and an auth token is detected, the Sentry SDK will
+   * automatically generate and upload source maps to Sentry during a production build.
+   *
+   * @default true
+   */
+  enabled?: boolean;
+
+  /**
+   * The auth token to use when uploading source maps to Sentry.
+   *
+   * Instead of specifying this option, you can also set the `SENTRY_AUTH_TOKEN` environment variable.
+   *
+   * To create an auth token, follow this guide:
+   * @see https://docs.sentry.io/product/accounts/auth-tokens/#organization-auth-tokens
+   */
+  authToken?: string;
+
+  /**
+   * The organization slug of your Sentry organization.
+   * Instead of specifying this option, you can also set the `SENTRY_ORG` environment variable.
+   */
+  org?: string;
+
+  /**
+   * The project slug of your Sentry project.
+   * Instead of specifying this option, you can also set the `SENTRY_PROJECT` environment variable.
+   */
+  project?: string;
+
+  /**
+   * If this flag is `true`, the Sentry plugin will collect some telemetry data and send it to Sentry.
+   * It will not collect any sensitive or user-specific data.
+   *
+   * @default true
+   */
+  telemetry?: boolean;
+
+  /**
+   * Options related to sourcemaps
+   */
+  sourcemaps?: {
```

Comment:
m: I find it a bit odd that we are nesting it like `sourceMapsUploadOptions.sourcemaps`. Can we hoist it?

Reply:
I found this a bit odd too but I'm following in the path of Nuxt and Sveltekit. I can change it but thought it might be better to align with the other packages, wdyt?

Reply:
Not sure I completely like how we did it in those packages tbh. I'll leave it up to you.

Reply:
I flattened this.